pkg/cmd/openshift-controller-manager/openshiftcontrolplane_default.go

```diff
@@ -3,14 +3,23 @@ package openshift_controller_manager
 import (
 	"time"
 
+	"k8s.io/klog"
+
+	configv1 "github.com/openshift/api/config/v1"
 	openshiftcontrolplanev1 "github.com/openshift/api/openshiftcontrolplane/v1"
 	"github.com/openshift/library-go/pkg/config/configdefaults"
 	"github.com/openshift/library-go/pkg/config/helpers"
 	leaderelectionconverter "github.com/openshift/library-go/pkg/config/leaderelection"
 )
 
 func setRecommendedOpenShiftControllerConfigDefaults(config *openshiftcontrolplanev1.OpenShiftControllerManagerConfig) {
-	configdefaults.SetRecommendedHTTPServingInfoDefaults(config.ServingInfo)
+	if config.ServingInfo == nil {
+		config.ServingInfo = &configv1.HTTPServingInfo{}
+		configdefaults.SetRecommendedHTTPServingInfoDefaults(config.ServingInfo)
+	} else if config.ServingInfo != nil && len(config.ServingInfo.BindAddress) == 0 {
+		klog.Warning("config.ServingInfo will be ignored as it contains an empty BindAddress")
+		config.ServingInfo = nil
+	}
```

Comment:
what is the goal of this block? When will either case happen?

Reply:
the idea was to make `config.ServingInfo` optional so that OCM doesn't expose `health` EP over SSL as this would require less step to run it on a local machine.

the first condition will be satisfied when there no `"servingInfo"` entry is in the configuration file
the second condition will be satisfied when there is empty `"servingInfo": {}` entry in the config file

Reply:
can you add a godoc one-liner for each?

Reply:
done - https://github.com/openshift/openshift-controller-manager/pull/30/commits/e6329e8c1f7618519b670ee96d33c09870b8c281